backend/controllers/cafeController.js

```diff
@@ -1,14 +1,47 @@
-const Cafe = require('../models/cafeModel')
+const Cafe = require('../models/cafeModel');
+const defaultDistance = 5000; // in meters, i.e., 5000 = 5km
 
 exports.getCafesByFacilitiesAndLocation = async (req, res, next) => {
-  const { facilities, requiredFacilities } = req.query;
-  const requestedFacilities = requiredFacilities ? requiredFacilities : facilities;
-  const requestedFacilitiesArray = requestedFacilities.split(',');
+  const { facilities, requiredFacilities, userLocation } = req.query;
+  const maxDistance = Number(req.query.distance) || defaultDistance;
+
+  if ((!requiredFacilities && !facilities) || !userLocation) {
```

Comment:
@pablisch Do we want to have the error handling for if no facilities in BE and FE? I have a modal for the instance of not selecting a facility, which would stop it from going to the BE altogether (the required facility part is still pending I guess). Just a thought, maybe double error handling is good on both sides though

Reply:
You're quite right, this should be handled in the frontend, but I think it is good practice to handle any possible errors in the backend too. Even though it is unlikely these errors will not be handled in the frontend, there is nothing to stop anyone making a manual browser request to the API causing an unhandled error and crashing the API server.

Reply:
As it goes, I handled this error really badly and the server would crash anyway. Thank you for highlighting the error handling 🙂